### PR TITLE
Introducing rubin-env

### PR DESCRIPTION
**Pull request by gcomoretto**
Also:

- Separated some common functions in bin/utils.sh
- Changed to miniconda version py38.4.9.2
- Removed -2 and -3 options from bin/deploy (usage, DM-19563)
- removed git and git-lfs deployment (now in rubin-env)
- if eups is not found in the environment (typical old environments) it creates an eups stand-alone env and it stacks the lsst-scipipe env on top.
- Saving pinned environment information (for src packages)
- Expanded rebuild Usage (DM-5461)
- added script bin/set_presreleased_env to set-up a rubin-env version from a fork, prior to its formal release to conda-forge

### CODE REVIEW

File: bin/deploy [hunk 1]
```diff
@@ -158,86 +116,8 @@ main() {
...
-  if [ -e "${lock_file}" ]; then
-    echo "::: conda lock file already present"
-  else
-    # if a branch or tag is provided, store the environment yaml inside the corresponding subfolder
-    mkdir -p "${lock_file%/*}"
-    $CURL "${CURL_OPTS[@]}" -# -L \
-       "${env_url}/${conda_lockfile}" \
-       --output "${lock_file}"
-  fi
+  # check wich platform we are running on
```
Comment (ktlim) [inline bin/deploy:119]:
```suggestion
  # check which platform we are running on
```

File: bin/deploy [hunk 2]
```diff
@@ -343,69 +202,17 @@ main() {
...
-      cd "$LSSTSW"
-      touch "${LSSTSW}/lfs/.git-lfs.deployed"
-    )
-  fi
+  # report packages in the current conda env
+  #
+  # note that `conda list -e`, which is the current format being used for the
+  # canoncial env files, doesn't seem to list pip installed packages.  The the
+  # "newer" yaml env file format from `conda env export` does include pip/pypi
+  # packages.
```
Comment (ktlim) [inline bin/deploy:210]: I don't think this comment is relevant anymore; it can be deleted.

File: bin/deploy [hunk 3]
```diff
@@ -416,45 +223,31 @@ main() {
...
-    fi
-
-    cd sources
-    if [[ -n $LSST_EUPS_GITREV ]]; then
-      git clone "$LSST_EUPS_GITREPO" eups
-      cd eups
-      git checkout "$LSST_EUPS_GITREV"
+  # backwards compatibility if EUPS was installed to a versioned directory (not from conda)
+  test ! -f "${LSSTSW}/eups/${LSST_EUPS_VERSION}/.deployed" || ( # EUPS
+    # in case eups fodler has already being moved to eups-legacy
```
Comment (ktlim) [inline bin/deploy:228]:
```suggestion
    # in case eups folder has already been moved to eups-legacy
```

File: bin/deploy [hunk 4]
```diff
@@ -416,45 +223,31 @@ main() {
...
-    make install
-    touch "${LSSTSW}/eups/${LSST_EUPS_VERSION}/.deployed"
   )
 
-  if [[ "$(readlink "${LSSTSW}/eups/current")" != "$LSST_EUPS_VERSION" ]]; then
-    echo "::: Making eups ${LSST_EUPS_VERSION} the default"
-    rm -f "${LSSTSW}/eups/current"
-    ln -s "${LSST_EUPS_VERSION}" "${LSSTSW}/eups/current"
+  # check if eups is not already available in the environment
+  # made it available in a separe eups-base environment
```
Comment (ktlim) [inline bin/deploy:240]:
```suggestion
  # if not, make it available from a separate eups-base environment
```

**Review by ktlim — APPROVED** [review]

File: bin/utils.sh [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+# function used to deploy environment
+deploy_scipipe_env() {
+  echo "---------------------  Deploying environment from scipipe_conda_env, git reference: ${LSST_SPLENV_REF}"
+
+  local conda_bleedfile="conda3_bleed-${pkg_postfix}.yml"
+  local conda_lockfile="conda-${pkg_postfix}.lock"
+
+  local lock_file="${LSSTSW}/env/${ENVREF}/${conda_lockfile}"
+
+  cd "$LSSTSW" || return
```
[the remark this answers is not in the document]
Comment (gcomoretto) [reply bin/utils.sh:66]: I just collected here the logic that was before in bin/deploy. Note that there is already a subshell in the second half of the function. Should I extend this subshell to all the function?

File: bin/set_prereleased_env [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+
+  log_file="conda_build_recipe.log"
+  echo "Conda Build Recipe. Redirecting to logfile: $log_file"
+  # tests seems to generate errors when building locally
+  # for the scope of this script we avoid to run test when executing conda build
+  conda build --no-test recipe/ 2>&1 > $log_file
+  echo "Recipe Built. Logfile: $log_file"
+
+  pkg_file=$(cat $log_file | grep "^anaconda upload" | awk '{ print $3 }')
+  # /Users/GComoretto/anaconda3/conda-bld/osx-64/rubinenv-0.1.0-0.tar.bz2
```
[the remark this answers is not in the document]
Comment (gcomoretto) [reply bin/set_prereleased_env:113]: to remove

File: bin/rebuild [hunk 7]
```diff
@@ -114,12 +112,6 @@ if [[ "$#" != "0" ]]; then
 fi
 
 (
-  # shellcheck disable=SC1090
-  # shellcheck disable=SC2154
-  . "${miniconda_path}/etc/profile.d/conda.sh"
-  # shellcheck disable=SC1091
-  conda activate "$LSST_CONDA_ENV_NAME"
```
[the remark this answers is not in the document]
Comment (gcomoretto) [reply bin/rebuild:121]: I added an extra line in the usage, and a check that fails if the environment is not active.

**Pull request merged**